internal/controller/dashboard_controller.go

```diff
@@ -0,0 +1,344 @@
+	DashboardTeamAnnotation = "giantswarm.io/organization"
+)
+
+func SetupDashboardReconciler(mgr manager.Manager, conf config.Config) error {
+	// Generate Grafana client
+	// Get grafana admin-password and admin-user
+	grafanaAdminCredentials := grafanaclient.AdminCredentials{
+		Username: conf.Environment.GrafanaAdminUsername,
+		Password: conf.Environment.GrafanaAdminPassword,
+	}
+	if grafanaAdminCredentials.Username == "" {
+		return fmt.Errorf("GrafanaAdminUsername not set: %q", conf.Environment.GrafanaAdminUsername)
+	}
+	if grafanaAdminCredentials.Password == "" {
+		return fmt.Errorf("GrafanaAdminPassword not set: %q", conf.Environment.GrafanaAdminPassword)
+	}
+
+	grafanaTLSConfig := grafanaclient.TLSConfig{
+		Cert: conf.Environment.GrafanaTLSCertFile,
+		Key:  conf.Environment.GrafanaTLSKeyFile,
+	}
+	grafanaAPI, err := grafanaclient.GenerateGrafanaClient(conf.GrafanaURL, grafanaAdminCredentials, grafanaTLSConfig)
+	if err != nil {
+		return fmt.Errorf("unable to create grafana client: %w", err)
+	}
+
+	r := &DashboardReconciler{
+		Client:     mgr.GetClient(),
+		Scheme:     mgr.GetScheme(),
+		GrafanaAPI: grafanaAPI,
+	}
+
+	err = r.SetupWithManager(mgr)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//+kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=core,resources=configmaps/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=core,resources=configmaps/finalizers,verbs=update
+
+// Reconcile is part of the main Kubernetes reconciliation loop which aims to
+// move the current state of the Dashboard closer to the desired state.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.16.0/pkg/reconcile
+func (r *DashboardReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	logger.Info("Started reconciling Grafana Dashboard Configmaps")
+	defer logger.Info("Finished reconciling Grafana Dashboard Configmaps")
+
+	dashboard := &v1.ConfigMap{}
+	err := r.Client.Get(ctx, req.NamespacedName, dashboard)
+	if err != nil {
+		return ctrl.Result{}, errors.WithStack(client.IgnoreNotFound(err))
+	}
+
+	// Handle deleted grafana dashboards
+	if !dashboard.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, r.reconcileDelete(ctx, dashboard)
+	}
+
+	// Handle non-deleted grafana dashboards
+	return r.reconcileCreate(ctx, dashboard)
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *DashboardReconciler) SetupWithManager(mgr ctrl.Manager) error {
+
+	labelSelectorPredicate, err := predicate.LabelSelectorPredicate(metav1.LabelSelector{MatchLabels: map[string]string{"app.giantswarm.io/kind": "dashboard"}})
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&v1.ConfigMap{}, builder.WithPredicates(labelSelectorPredicate)).
```

Comment:
Adding a name to the controller otherwise it default to the reconciled resources, so here it would be name configmap in the logs.
```suggestion
	return ctrl.NewControllerManagedBy(mgr).
		Named("dashboards")
		For(&v1.ConfigMap{}, builder.WithPredicates(labelSelectorPredicate)).
```